streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -270,11 +270,12 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
         logInfo("Count = " + count + ", Running count = " + runningCount)
       }
       ssc.start()
-      ssc.awaitTerminationOrTimeout(500)
+      eventually(timeout(10.seconds), interval(10.millis)) {
+        assert(runningCount > 0)
+      }
       ssc.stop(stopSparkContext = false, stopGracefully = true)
       logInfo("Running count = " + runningCount)
       logInfo("TestReceiver.counter = " + TestReceiver.counter.get())
-      assert(runningCount > 0)
       assert(
         (TestReceiver.counter.get() == runningCount + 1) ||
           (TestReceiver.counter.get() == runningCount + 2),
```

Comment:
Could you try running it 100 times on your machine?


On Fri, Jul 31, 2015 at 1:55 AM, Shixiong Zhu <notifications@github.com>
wrote:

> In
> streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala
> <https://github.com/apache/spark/pull/7823#discussion_r35956779>:
>
> >        ssc.stop(stopSparkContext = false, stopGracefully = true)
> >        logInfo("Running count = " + runningCount)
> >        logInfo("TestReceiver.counter = " + TestReceiver.counter.get())
> > -      assert(runningCount > 0)
> >        assert(
> >          (TestReceiver.counter.get() == runningCount + 1) ||
> >            (TestReceiver.counter.get() == runningCount + 2),
>
> Okey. I think it's impossible to be true so I'm going to remove it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/7823/files#r35956779>.
>